v4/newrelic/segments.go

```diff
@@ -196,6 +201,7 @@ func (s *Segment) End() {
 		return
 	}
 	if s.StartTime.isEnded() {
+		logAlreadyEnded(s.StartTime, s.Name)
```

Comment:
A note on why I use a function here rather than just adding a logging method on the segment: there are a bunch of different types of segments, and I didn't want to add a method to each one. Also on each of these segment types the logger is pretty nested (`s.StartTime.span.thread.logger`), and most of those things along the path can be `nil` but I didn't want to check that whole path explicitly for every segment type.

Each segment type has a `SegmentStartTime` field which can't be `nil` because it's a struct, so that's why I chose `s.StartTime` to pass in to the method.

Reply:
Makes sense. If these were exported fields it would matter a whole lot more about where store the logger reference (and whether we use a function here vs add methods to the segment). Thanks for checking on whether StartTime can be `nil`!